core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -28,60 +28,148 @@
   }
 
   /**
    * Calls the given {@link RpcCallable} and handles any exceptions thrown.
    *
-   * The logAnyFailure parameter indicates whether or not AlluxioExceptions are expected results
-   * (for example it would be false for the exists() call). In this case, we do not log the failure
-   * or increment failure metrics. When a RuntimeException is thrown, we always treat it as a
-   * failure and log an error and increment metrics.
+   * The failureOk parameter indicates whether or not AlluxioExceptions are expected results (for
+   * example it would be false for the exists() call). In this case, we do not log the failure or
+   * increment failure metrics. When a RuntimeException is thrown, we always treat it as a failure
+   * and log an error and increment metrics.
    *
    * @param logger the logger to use for this call
    * @param callable the callable to call
-   * @param logAnyFailure whether to log or increment metrics whenever the RPC fails
+   * @param methodName the name of the method, used for metrics
+   * @param failureOk whether failures are expected (affects logging and metrics)
+   * @param description the format string of the description, used for logging
+   * @param args the arguments for the description
    * @param <T> the return type of the callable
    * @return the return value from calling the callable
    * @throws AlluxioTException if the callable throws an exception
    */
-  public static <T> T call(Logger logger, RpcCallable<T> callable, boolean logAnyFailure)
-      throws AlluxioTException {
-    try (Timer.Context ctx = MetricsSystem.timer(getQualifiedMetricName(callable)).time()) {
-      logger.debug("Enter: {}", callable);
+  public static <T> T call(Logger logger, RpcCallable<T> callable, String methodName,
+      boolean failureOk, String description, Object... args) throws AlluxioTException {
+    // avoid string format for better performance if debug is off
+    String debugDesc = logger.isDebugEnabled() ? String.format(description, args) : null;
+    try (Timer.Context ctx = MetricsSystem.timer(getQualifiedMetricName(methodName)).time()) {
+      logger.debug("Enter: {}: {}", methodName, debugDesc);
       T ret = callable.call();
-      logger.debug("Exit (OK): {}", callable);
+      logger.debug("Exit (OK): {}: {}", methodName, debugDesc);
       return ret;
     } catch (AlluxioException e) {
-      logger.debug("Exit (Error): {}", callable, e);
-      if (logAnyFailure) {
-        MetricsSystem.counter(getQualifiedFailureMetricName(callable)).inc();
+      logger.debug("Exit (Error): {}: {}", methodName, debugDesc, e);
+      if (!failureOk) {
+        MetricsSystem.counter(getQualifiedFailureMetricName(methodName)).inc();
         if (!logger.isDebugEnabled()) {
-          logger.warn("{}, Error={}", callable, e.getMessage());
+          logger.warn("{}: {}, Error={}", methodName, String.format(description, args), e);
```

Comment:
@calvinjia should this be `e` or `e.getMessageIO`?

Reply:
I think we want `e.toString` because that prints both the exception type and message.

Reply:
make sense. just to confirm that the output will be different after this PR.